test/Main.purs

```diff
@@ -0,0 +1,128 @@
+module Test.Main where
```

Comment:
Yeah, although that's pretty much a search and replace job. Once we decide on the new `Semiring` subclass laws I'll update it and create both versions.